main.py

```diff
@@ -221,22 +225,49 @@ def _set_secret(config_filename):
     Secret.set(secret)
 
 
+def _start_app_force_kill_countdown(seconds):
+    """
+    Start a daemon thread that will wait the specified number of seconds and then dump debug info and forcefully kill
+    the app. Note that this hard kill will only be executed if the app is unexpectedly hanging. Under normal
+    circumstances, the app will exit cleanly (all nondaemon threads will finish) before the specified delay has elapsed.
```

Comment:
Are you saying here that we call an explicit exit() of some sort in python in the subcommand class for the master/slave services?